src/client/automation/playback/click/click-command.js

```diff
@@ -0,0 +1,142 @@
+    constructor (eventState, eventArgs) {
+        super(eventState, eventArgs);
+    }
+
+    run () {
+        if (this.eventState.clickElement && browserUtils.isFirefox)
+            this._bindClickHandler(this.eventState.clickElement);
+
+        super.run();
+    }
+
+    _bindClickHandler (element) {
+        const onclick = e => {
+            eventUtils.preventDefault(e, true);
+            eventUtils.unbind(element, 'click', onclick);
+        };
+
+        eventUtils.bind(element, 'click', onclick);
+    }
+}
+
+class SelectElementClickCommand extends ElementClickCommand {
+    constructor (eventState, eventArgs) {
+        super(eventState, eventArgs);
+    }
+
+    run () {
+        super.run();
+
+        this._toggleSelectOptionList();
+    }
+
+    _toggleSelectOptionList () {
+        // NOTE: Emulating the click event on the 'select' element doesn't expand the
+        // dropdown with options (except chrome), therefore we should emulate it.
+        const element              = this.eventArgs.element;
+        const isSelectWithDropDown = styleUtils.getSelectElementSize(element) === 1;
+
+        if (isSelectWithDropDown && this.eventState.simulateDefaultBehavior !== false) {
+            if (selectElementUI.isOptionListExpanded(element))
+                selectElementUI.collapseOptionList();
+            else
+                selectElementUI.expandOptionList(element);
+        }
+    }
+}
+
+class OptionElementClickCommand extends ElementClickCommand {
+    constructor (eventState, eventArgs) {
+        super(eventState, eventArgs);
+    }
+
+    run () {
+        return this.eventArgs.element;
+    }
+}
+
+class LabelledCheckboxElementClickCommand extends ElementClickCommand {
+    constructor (eventState, eventArgs) {
+        super(eventState, eventArgs);
+
+        this.checkbox = getElementBoundToLabel(this.eventArgs.element);
+    }
+
+    run () {
+        let changed = false;
+
+        const onChange = () => {
+            changed = true;
+        };
+
+        listeners.addInternalEventListener(window, ['change'], onChange);
+
+        super.run();
+
+        listeners.removeInternalEventListener(window, ['change'], onChange);
+
+        if (browserUtils.isChrome && !changed)
+            this._ensureCheckboxStateChanged();
+    }
+
+    _ensureCheckboxStateChanged () {
+        this.checkbox.checked = !this.checkbox.checked;
+
+        eventSimulator.change(this.checkbox);
+    }
+}
+
+export default function (eventState, eventArgs) {
+    const elementBoundToLabel = getElementBoundToLabel(eventArgs.element);
+    const isSelectElement     = domUtils.isSelectElement(eventArgs.element);
+    const isOptionElement     = domUtils.isOptionElement(eventArgs.element);
+    const isColorInputElement = domUtils.isColorInputElement(eventState.clickElement);
+    const isLabelledCheckbox  = elementBoundToLabel && domUtils.isCheckboxElement(elementBoundToLabel);
+
+    if (isSelectElement)
```

Comment:
if (isSelectElement)...
else if (isOptionElement)...
else if (...)
else ...

Reply:
for what?

Reply:
Sorry, I've missed the `return` statement. All is OK.